features/126-record-id/implementation.md

```diff
@@ -0,0 +1,210 @@
+
+// New, no longer support deleting multiple types
+- (void)deleteRecordsWithIDs:(NSArray<NSString *> *)recordIDs
+                        type:(NSString *)recordType
+           completionHandler:(void (^)(NSArray<NSString *> *deletedRecordIDs,
+                                       NSError *error))completionHandler
+       perRecordErrorHandler:(void (^)(NSString *recordID,
+                                       NSError *error))errorHandler;
+```
+
+The record save API is unchanged.
+
+The Record class will be changed as follows:
+
+```obj-c
+// Deprecated
+@property (nonatomic, readonly, copy) NSString *recordName;
+
+// Old
+@property (nonatomic, readonly, copy) SKYRecordID *recordID;
+@property (nonatomic, readonly, copy) NSString *recordType;
+
+// New
+@property (nonatomic, readonly, copy) NSString *recordID;
+@property (nonatomic, readonly, copy) NSString *recordType;
+
+// Deprecated
++ (instancetype)recordWithRecordID:(SKYRecordID *)recordId
+                              data:(NSDictionary<NSString *, id> *_Nullable)data;
+
+// Old
++ (instancetype)recordWithRecordType:(NSString *)recordType name:(NSString *)recordName;
++ (instancetype)recordWithRecordType:(NSString *)recordType
+                                name:(NSString *)recordName
+                                data:(NSDictionary<NSString *, id> *_Nullable)data;
+
+// New
++ (instancetype)recordWithRecordType:(NSString *)recordType recordID:(NSString *)recordID;
++ (instancetype)recordWithRecordType:(NSString *)recordType
+                            recordID:(NSString *)recordID
+                                data:(NSDictionary<NSString *, id> *_Nullable)data;
+```
+
+The Reference class will be changed as follows:
+
+```obj-c
+// Old
+@property (nonatomic, readonly, copy) SKYRecordID *recordID;
+
+// New
+@property (nonatomic, readonly, copy) NSString *recordID;
+@property (nonatomic, readonly, copy) NSString *recordType;
+```
+
+The Reference class will be changed as follows:
+
+```obj-c
+// Old
+@property (nonatomic, readonly, copy) SKYRecordID *recordID;
+
+// New
+@property (nonatomic, readonly, copy) NSString *recordID;
+@property (nonatomic, readonly, copy) NSString *recordType;
+
+// Old
++ (instancetype)referenceWithRecordID:(SKYRecordID *)recordID;
+
+// New
++ (instancetype)referenceWithRecordType:(NSString *)recordType
+                               recordID:(NSString *)recordID;
+```
+
+## Android SDK
+
+The SDK does not expose deprecated ID in the API. The Android SDK does not
+support fetching record by ID (need to use query API) and deletion requires
+Record object.
+
+## JS SDK
+
+The `id` property of Record class will return Record ID instead of deprecated
+ID.
+
+```javascript
+var Note = Record.extend('note');
+var note = new Note({});
+
+// Old
+console.log(note.id);  // 'note/91c02d33-ea8a-41d3-ae6c-a87e064eaa5c'
+console.log(note.recordType);  // 'note'
+
+// New
+console.log(note.recordID);  // '91c02d33-ea8a-41d3-ae6c-a87e064eaa5c'
```

Comment:
Is it intended to rename `id` to `recordID`? Or it is an unconscious choice?

Reply:
It is renamed to recordID to keep consistency with other platforms. Android SDK has `recordID` as well.